File: PATTERNS/Pattern47/Pattern47.java
import java.util.Scanner;

public class Pattern47 {
    public static void main(String[] args) {
        System.out.print("Enter the number of rows : ");
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        for (int i = 1; i <= n; i++) {
            for (int j = n; j >= 1; j--) {
                if (j == i)
                    System.out.print("*");
                else
                    System.out.print(j);
            }
            System.out.println();
        }
        sc.close();

    }
}

File: PATTERNS/Pattern47/Pattern47.js
function generatePattern(input) {
    let pattern = '';
    // let input = prompt("enter a number: ");

    //adding an example number since the prompt function isn't working
  
    for (let i = 1; i<=input; i++) {
      for (let j = input; j>=1; j--) {
        if (j === i) {
          pattern += '*';
        } else {
          pattern += j;
        }
      }
      pattern += '\n';
    }
  
    return pattern;
  }


//readline is an object, we use the built in require function in js to fetch the readline module

var readline = require('readline'); 
var rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

rl.question('pick a number ', (n) => {
  
    var pattern = generatePattern(parseInt(n));
    console.log(pattern);
    rl.close();
  
});

